src/ui.text-alert/index.vue

```diff
@@ -1,30 +1,55 @@
 <template>
   <div v-if="isShownAlert" :data-cy="dataCy" v-bind="wrapperAttrs">
-    <div v-bind="bodyAttrs">
-      <!-- @slot Use it to add something inside. -->
-      <slot />
-
-      <div v-if="!hasSlotContent($slots['default'])" v-html="html" />
+    <div>
+      <!-- @slot Use it to add some component above text. -->
+      <slot name="top" />
     </div>
-
-    <UButton
-      v-if="closeIcon"
-      size="sm"
-      variant="thirdary"
-      :color="color"
-      square
-      v-bind="buttonAttrs"
-    >
-      <UIcon
-        internal
-        size="xs"
+    <div v-bind="bodyAttrs">
+      <div>
+        <!-- @slot Use it to add some component before text. -->
+        <slot name="left" />
+      </div>
+      <div>
+        <div v-bind="titleAttrs">
+          <slot name="title" :title="title">
+            <div v-if="title" v-bind="titleAttrs" v-text="title" />
+          </slot>
+        </div>
+        <div v-bind="descriptionAttrs">
+          <slot name="description" :description="description">
+            <div v-if="description" v-bind="descriptionAttrs" v-text="description" />
+          </slot>
+        </div>
+        <slot />
+        <div v-if="!hasSlotContent($slots.default)" v-html="html" />
+      </div>
+      <UButton
+        v-if="closable"
+        size="sm"
+        variant="thirdary"
         :color="color"
-        :name="config.iconName"
-        :data-cy="`${dataCy}-button`"
-        v-bind="iconAttrs"
-        @click="onHidden"
-      />
-    </UButton>
+        square
+        v-bind="buttonAttrs"
+      >
+        <UIcon
+          internal
+          size="xs"
+          :color="color"
+          :name="config.iconName"
+          :data-cy="`${dataCy}-button`"
+          v-bind="iconAttrs"
+          @click="onHidden"
+        />
+      </UButton>
+      <div>
+        <!-- @slot Use it to add some component after text. -->
+        <slot name="right" />
+      </div>
+    </div>
+    <div v-bind="wrapperAttrs.vertical">
```

Comment:
`wrapperAttrs.vertical` - the construction is wrong. Should be 1 level variable.